src/attestation_types/quote/mod.rs

```diff
@@ -79,3 +88,70 @@ pub struct Quote {
     /// supporting data.
     sig_data: SigData,
 }
+
+// The size of the Quote is not known at compile time. It is specified in the data itself.
+impl TryFrom<&[u8]> for Quote {
+    type Error = QuoteError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        let mut header = [0u8; QUOTE_HEADER_LEN];
+        header.copy_from_slice(&bytes[0..QUOTE_HEADER_LEN]);
```

Comment:
would it panic if bytes is too small? Does it need a length check, which error out early?

Reply:
Thanks, very good point.